src/engine/controls/bpmcontrol.cpp

```diff
@@ -83,6 +83,10 @@ BpmControl::BpmControl(QString group,
     connect(m_pTranslateBeatsLater, &ControlObject::valueChanged,
             this, &BpmControl::slotTranslateBeatsLater,
             Qt::DirectConnection);
+    m_pBeatsSetBarBeat = new ControlPushButton(ConfigKey(group, "beats_set_bar_beat"), false);
```

Comment:
What you say makes a lot of sense. I'm not Qt literate, so for this part I just copied the code for another button in the same file without trying yo understand too much.
If this must be modified, for coherence we must also fix all the others. Having some of them looking different than the others can be quite confusing for poor little guys like trying to understand how to develop Mixxx.
I can do the work in the file I'm editing or, if needed I can volunteer to prepare a separate PR to fix them, no to entangle this PR.
Out of curiosity, did you notice just looking at the code or by some message in the logs??

Reply:
I just have read the code. 

You are right with the neighbor code. Decide yourself, just add a delete in the destructor, or remove all of them by using make_unique for all. I don't really mind.

Reply:
I will go for unique_ptr. AFAIK it's a more robust approach.

Reply:
I have found that ControlPushButton can't be made unique_ptr. It's missing a copy constructor. I will prefer not to modify a class that is outside the scope of my changes. I feel it's dangerous because I have not a fully understanding of what are that objects. So I will go for the delete approach.
ControlProxy can be made unique, so I can do the changes to enforce the no naked new rule.
Anyway, if we really prefer not to have naked news, someone with better understanding that me must take a look at them at some point.

Reply:
> I have found that ControlPushButton can't be made unique_ptr. It's missing a copy constructor.

I'm not aware that objects managed by unique_ptr require a copy constructor? But unique_ptr itself is not copyable, only moveable.

Reply:
Using unique_ptr is the recommended approach for non-QObject, internal, managed pointers in classes that don't need to be copyable like BpmControl.

Reply:
Done